lib/iron_bank/faraday_middleware.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+# IronBank main module
+module IronBank
+  # IronBank Faraday middleware module
+  module FaradayMiddleware
+    autoload :RaiseError,    'faraday_middleware/raise_error'
```

Comment:
not a fan of `autoload`, let's remove it and instead add it to the list of `require` in `iron_bank.rb`

cf. https://bugs.ruby-lang.org/issues/5653

> "I strongly discourage the use of autoload in any standard libraries" (Re: autoload will be dead)
(Matz)

Reply:
My bad, I forgot to remove after adding the `require` inside the main module